src/System.Private.Uri/tests/UnitTests/IriEscapeUnescapeTest.cs

```diff
@@ -408,15 +401,43 @@ public void ValidatePadding()
             {
                 for (int i = 0; i < _memblock.Length; i++)
                 {
-                    if ((i < padding) || (i >= padding + _len))
+                    AssertValidPadding(i, _len, _memblock[i]);
+                }
+            }
+
+            public static void ValidatePadding(ValueStringBuilder pooledArray)
```

Comment:
I don't understand why you need to pass down a ValueStringBuilder here?

Reply:
The method validates that all the result characters are written into the expected range inside the "pooledArray".